transcoders.go

```diff
@@ -293,12 +294,22 @@ func garlic32Validate(b []byte) error {
 var TranscoderP2P = NewTranscoderFromFunctions(p2pStB, p2pBtS, p2pVal)
 
 func p2pStB(s string) ([]byte, error) {
-	// the address is a varint prefixed multihash string representation
+	// check if the address is a varint prefixed multihash string representation
 	m, err := mh.FromB58String(s)
+	if err == nil {
+		return m, nil
+	}
+
+	// check if the address is a CID
+	c, err := cid.Decode(s)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse p2p addr: %s %s", s, err)
 	}
-	return m, nil
+	if c.Version() > 0 && c.Type() != cid.Libp2pKey {
```

Comment:
`c.Version()` should be 1, not o.

Reply:
I have `c.Version() > 0` here. I guess we don't know what CIDv2+ will look like anyway so I can just assert `c.Version() == 1` though.

Reply:
I decided to just ignore the Version field entirely as we do in go-libp2p-core. As far as I can tell there is literally no way (except perhaps with reflection) to create a CID with version number not equal to 0 or 1.

Lmk if there are any other checks you think are missing here.

Reply:
Ah, nvm, now I get it.

> I decided to just ignore the Version field entirely as we do in go-libp2p-core. As far as I can tell there is literally no way (except perhaps with reflection) to create a CID with version number not equal to 0 or 1.

Checking the type should be sufficient.